test/common/network/utility_test.cc

```diff
@@ -173,7 +173,7 @@ TEST(NetworkUtility, LocalConnection) {
 
   testing::NiceMock<Network::MockConnectionSocket> socket;
 
-  EXPECT_CALL(socket, remoteAddress()).WillRepeatedly(testing::ReturnRef(local_addr));
+  EXPECT_CALL(socket, localAddress()).WillRepeatedly(testing::ReturnRef(local_addr));
```

Comment:
This is a glitch in the unit tests (returning local_addr for calls to remoteAddress()) that obfuscated the fact that isLocalConnection() was accidentally doing pointer comparisons.